### DISCUSSION TITLE
APB3Decoder

### DISCUSSION THREAD

**Discussion by ICJJ** [SpinalHDL/SpinalHDL#1474] [Q&A]
https://github.com/SpinalHDL/SpinalWorkshop/blob/e7ebcf1cec36e3392512cf27a928fdb650cd517e/src/main/scala/workshop/apb3decoder/Apb3Decoder.scala#L32
Why apb3Decoder has more than one master and one slave？
It should be and only one master,that is apbBridge,and more slave according to AMBA specs.
It confuse me.

**Comment by Dolu1990** [comment]
master / slave words are confusing words as it depend from which perspective you look at them they mean the oposite.

In SpinalHDL their meaning is from the perspective of the component they are in.
slave => you receive commands through this bus
master => you give commands through this bus
